script.js

```diff
@@ -1,4 +1,193 @@
+//global variables
+let playerTurnCounter = 1;
+const playerDice = [];
+let totalPlayer = 0;
+let numOfDice = 0;
+let playerScore = [];
+let autoModeFlag = 1;
+let leaderBoard = [];
+let reverseModeFlag = 0;
+let knockOutFlag = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (input.toLowerCase().includes("reverse")) {
+    reverseModeFlag = 1;
+    return "Reversed mode activated. Please get the lowest score to win";
+  }
+  if (input.toLowerCase().includes("knockout")) {
+    knockOutFlag = 1;
+    return "Knockout mode activated. Please enter 3 or more players to play";
+  }
+  if (input.toLowerCase().includes("reset")) {
+    playerTurnCounter = 1;
+    totalPlayer = 0;
+    numOfDice = 0;
+    autoModeFlag = 1;
+    reverseModeFlag = 0;
+    knockOutFlag = 0;
+    return "Game reset. Enter number of players to play";
+  }
+
+  if (totalPlayer == 0) {
+    totalPlayer = input;
+    for (let i = 0; i < totalPlayer; i++) {
+      playerDice[i] = [];
+      playerScore[i] = 0;
+    }
+    return `${totalPlayer} players will be playing the game. Please enter the number of dice to roll.`;
+  }
+  if (numOfDice == 0) {
+    numOfDice = input;
+    return `${numOfDice} dice will be rolled for this game. Player 1 please press submit to roll`;
+  }
+  if (knockOutFlag == 1) {
+    return knockOut(input);
+  }
+  return normalGame();
+};
+
+//roll dice
+var diceRoll = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+//rolls the number of dice for current player and stores them in a dice array after sorting big to small
+var playerTurn = function (noOfDice, playerTurnCounter) {
+  for (let i = 0; i < noOfDice; i++) {
+    playerDice[playerTurnCounter - 1][i] = diceRoll();
+  }
+  if (reverseModeFlag == 1) {
+    playerDice[playerTurnCounter - 1].sort();
+  } else {
+    playerDice[playerTurnCounter - 1].sort().reverse();
+  }
+};
+
+//converts the dice roll into the largest number possible and stores them in a score array
+var autoMode = function (diceArray) {
+  playerScore[playerTurnCounter - 1] += parseInt(
+    diceArray.toString().replace(/,/g, "")
+  );
+};
+
+// reset global variables to reset the game
+var resetGame = function () {
+  //totalPlayer = 0;
+  numOfDice = 0;
+  playerTurnCounter = 1;
+};
+
+// to create a leaderboard of the player scores by initalising a new array of arrays that carry the player's score and index number then sorting that array according to gamemode requirements
+var scoreBoard = function () {
+  for (let i = 0; i < totalPlayer; i++) {
+    leaderBoard[i] = [];
+  }
+  for (let i = 0; i < totalPlayer; i++) {
+    leaderBoard[i] = [playerScore[i], [i + 1]];
```

Comment:
You can put line 86 in line 83.

Reply:
i had problems with using line 86 first, it ran into an error where it couldnt assign the player score into the array as it didnt exist. So i thought i had to initialise the array of arrays first before i could do the assignment. could i have made line 86 work by itself?

Reply:
Yup I just tried replacing line 83 with 86 and deleting 85-87 and it works :)